### ISSUE TITLE
Failed to compile

### ISSUE THREAD

**Issue by npinaeva** [michaelknoch/ng2-cytoscape#6]
Tried to use updated version, installed it with `npm install` and did everything like in readme. Got compilation error in `import { NgCytoscapeModule } from 'ng2-cytoscape/dist'`:

Failed to compile.
/home/c2-gui/src/app/app.module.ts (20,10): Module '"/home/c2-gui/node_modules/ng2-cytoscape/dist/index"' has no exported member 'NgCytoscapeModule'.

Tried to change it to `import { NgCytoscapeModule } from 'ng2-cytoscape'` and got another error:

![2017-06-14 19-29-43](https://user-images.githubusercontent.com/15113699/27181818-e6345ba4-51e1-11e7-8117-aaf630619c36.png)

I will appreciate if you find out what i'm doing wrong.

**Comment by tesoknyra** [comment]
I have the same problem. Anyone, Can help?

**Comment by tesoknyra** [comment]
I meet the same problem. Anyone, Can help?

**Comment by tesoknyra** [comment]
I meet the same problem. Anyone, Can help?

**Comment by michaelknoch** [comment]
does this problem exist when you roll back to previous version 0.1.5?
@tesoknyra @npinaeva

**Comment by npinaeva** [comment]
Yes, the same compilation error, but I didn't find any occurrences of NgCytoscapeModule in previous version at all.

**Comment by tesoknyra** [comment]
My problem is solved because i just add script in .angular-cli.json such:
"scripts": ["../node_modules/cytoscape/dist/cytoscape.min.js",

**Comment by manikob** [comment]
Hi,
when I use import like this:
`import {NgCytoscapeModule} from "ng2-cytoscape";`
I've got similar error:

```
ERROR in Error encountered resolving symbol values statically. Function calls are not supported. Consider replacing the function or la
mbda with a reference to an exported function (position 194:50 in the original .ts file), resolving symbol NgModule in C:/SOURCES/AG4/
GDashboard/node_modules/ng2-cytoscape/node_modules/@angular/core/core.d.ts, resolving symbol NgCytoscapeModule in C:/SOURCES/AG4/GDash
board/node_modules/ng2-cytoscape/ng2-cytoscape.ts, resolving symbol NgCytoscapeModule in C:/SOURCES/AG4/GDashboard/node_modules/ng2-cy
toscape/ng2-cytoscape.ts
```

Do you have any idea what is wrong? Am I missing something?

**Comment by anvial** [comment]
So, does anyone find the solution of the problem?

**Comment by michaelknoch** [comment]
i would be very glad if someone can provide a fix for the issue and submit a pr. I'm currently very busy writing react native and swift code so I'm a little out of business about angular2, typescript and typings. @anvial @manikob

**Comment by ash2014** [comment]
Was anyone able to fix the "Cytoscape is not defined error" ? I tried installing cytoscape as dependency but I see the same error. Please update if anyone figured out the solution.

**Comment by anvial** [comment]
I just use the code of @michaelknoch module and injected it in own project as a ngModule. And it Works! =) 

Sorry for dropbox example. 
https://www.dropbox.com/s/i5r97zm599lpu3j/ng-cytoscape.zip?dl=0

If I have time I will prepare the plunker example.

**Comment by ash2014** [comment]
@anvial what does your package.json look like ?

**Comment by anvial** [comment]
Here it is:

{
  ...
  "dependencies": {
    "angular/animations": "^4.1.2",
    "angular/cdk": "^2.0.0-beta.8",
    "angular/common": "^4.0.0",
    "angular/compiler": "^4.0.0",
    "angular/core": "^4.0.0",
    "angular/flex-layout": "^2.0.0-beta.8",
    "angular/forms": "^4.0.0",
    "angular/http": "^4.0.0",
    "angular/material": "^2.0.0-beta.5",
    "angular/platform-browser": "^4.0.0",
    "angular/platform-browser-dynamic": "^4.0.0",
    "angular/router": "^4.0.0",
    ...
    "cytoscape": "^3.1.3",
    "cytoscape-cola": "^1.6.0",
    "cytoscape-cxtmenu": "^2.10.3",
    "cytoscape-edgehandles": "^2.13.1",
    "cytoscape-panzoom": "^2.5.2",
    "cytoscape.js-undo-redo": "^1.0.1",
    ...
  },
  "devDependencies": {
    "angular/cli": "1.2.1",
    "angular/compiler-cli": "^4.0.0",
    "types/jasmine": "2.5.53",
    "types/node": "~8.0.1",
    "codelyzer": "~3.1.1",
    "jasmine-core": "~2.6.3",
    "jasmine-spec-reporter": "~4.1.0",
    "karma": "~1.7.0",
    "karma-chrome-launcher": "~2.2.0",
    "karma-cli": "~1.0.1",
    "karma-coverage-istanbul-reporter": "^1.3.0",
    "karma-jasmine": "~1.1.0",
    "karma-jasmine-html-reporter": "^0.2.2",
    "protractor": "~5.1.0",
    "ts-node": "~3.2.0",
    "tslint": "~5.5.0",
    "typescript": "~2.3.4"
  }

**Comment by ash2014** [comment]
Thanks @anvial , so you did not do npm install ng2-cytoscape , but just use the original cytoscape library and componentize it ?

**Comment by anvial** [comment]
Yeap, to avoid installation problems.

**Comment by ssuperczynski** [comment]
I was able to reproduce above error, and fix it.

1. When you run npm install, go to node_modules/ng2-cytoscape and check node_modules. If @angular is listed, it could cause an error, since it should use global node_modules, not ng2-cytoscape one.
2. when you use angular-cli add these lines to angular-cli.json:
```
"scripts": [
        "../node_modules/jquery/dist/jquery.js",
        "../node_modules/cytoscape/dist/cytoscape.js"
      ],
```
3. Be sure ng2-cytoscape has the same version of angular, that your project.
My project has version `^4.3.5` since ng2-cytoscape has `4.2.2` and it was causing an error too. I made pull request to fix this issue too https://github.com/michaelknoch/ng2-cytoscape/pull/7

**Comment by ssuperczynski** [comment]
I just submitted pull request to fix all the errors mentioned above
https://github.com/michaelknoch/ng2-cytoscape/pull/8

Since it is not merged try to test my fork, by including this line to package json to dependencies:
`"ng2-cytoscape": "https://github.com/ssuperczynski/ng2-cytoscape#master",`

**Comment by michaelknoch** [comment]
its merged and i published 0.5.0 on npm some seconds ago. thanks @ssuperczynski for your help!
